code/modules/surgery/organs/augments_internal.dm

```diff
@@ -303,6 +303,134 @@
+		if(prob(100 / severity) && owner)
+			for(var/datum/action/item_action/organ_action/toggle/sensory_enhancer/SE in owner.actions)
+				SE.Trigger(FALSE, TRUE, TRUE)
+	. = ..()
+
+/datum/action/item_action/organ_action/toggle/sensory_enhancer
+	name = "Activate Qani-Laaca System"
+	desc = "Activates your Qani-Laaca computer and grants you its powers. LMB: Short, safer activation. ALT/MIDDLE: Longer, more powerful, more dangerous activation."
+	button_icon = 'icons/obj/surgery.dmi'
+	button_icon_state = "sandy"
+	check_flags = AB_CHECK_CONSCIOUS
+	/// Keeps track of how much mephedrone we inject into people on activation
+	var/injection_amount = 10
+
+
+/datum/action/item_action/organ_action/toggle/sensory_enhancer/AltTrigger()
+	Trigger(FALSE, TRUE)
+
+/datum/action/item_action/organ_action/toggle/sensory_enhancer/Trigger(left_click, attack_self, emp_triggered = FALSE)
+	. = ..()
+	if(istype(target, /obj/item/organ/internal/cyberimp/brain/sensory_enhancer))
+		var/obj/item/organ/internal/cyberimp/brain/sensory_enhancer/ourtarget = target
+		if(!COOLDOWN_FINISHED(ourtarget, sensory_enhancer_cooldown))
+			to_chat(owner, "<span class='warning'>[ourtarget] is still on cooldown for another [round(COOLDOWN_TIMELEFT(ourtarget, sensory_enhancer_cooldown), 1 SECONDS) / 10] seconds!</span>")
+			return
+
+		COOLDOWN_START(ourtarget, sensory_enhancer_cooldown, 5 MINUTES)
+
+		injection_amount = 10
+
+		if(!left_click)
+			injection_amount = 20
+		if(emp_triggered)
+			injection_amount = 40 //Time for a quick medical visit
+		Activate()
+
+
+/datum/action/item_action/organ_action/toggle/sensory_enhancer/proc/Activate(atom/target)
+
+	var/mob/living/carbon/human/human_owner = owner
+
+	human_owner.reagents.add_reagent("mephedrone", injection_amount)
+
+	owner.visible_message("<span class='danger'>[owner.name] jolts suddenly as two small glass vials are fired from ports in the implant on [owner.p_their()] spine, shattering as they land.</span>", \
+			"<span class='userdanger'>You jolt suddenly as your Qani-Laaca system ejects two empty glass vials rearward, shattering as they land.</span>")
+	playsound(human_owner, 'sound/goonstation/items/hypo.ogg', 80, TRUE)
+
+	var/obj/item/telegraph_vial = new /obj/item/qani_laaca_telegraph(get_turf(owner))
+	var/turf/turf_we_throw_at = get_edge_target_turf(owner, REVERSE_DIR(owner.dir))
+	telegraph_vial.throw_at(turf_we_throw_at, 5, 1)
+
+	// Safety net in case the injection amount doesn't get reset. Apparently it happened to someone in a round.
+	injection_amount = initial(injection_amount)
+
+/obj/item/qani_laaca_telegraph
+	name = "spent Qani-Laaca cartridge"
+	desc = "A small glass vial, usually kept in a large stack inside a Qani-Laaca implant, that is broken open and ejected \
+		each time the implant is used. If you're looking at one long enough to think about it this long, you either have fast eyes \
+		or were lucky enough to catch onechem_heater  before it broke."
```

Comment:
`onechem_heater`?

Reply:
lmao
find and replace moment

Reply:
actually probably just a bad ctrl v
not even a replace